### ISSUE TITLE
Latest build fails under NetBSD-current aarch64 and amd53

### ISSUE THREAD

**Issue by ci4ic4** [ziglang/zig#18119]
### Zig Version

0.12.0-dev.1736+2252dcc50

### Steps to Reproduce and Observed Behavior

Standard zig build procedure using self-built llvm-17, building using ninja:

```
git pull
cd build
cmake -DCMAKE_BUILD_TYPE=Release -DZIG_NO_LIB=ON -GNinja ..
ninja install
```

Observed behaviour:

```
$ ninja install
[6/9] Building C object CMakeFiles/zig2.dir/zig2.c.o
/home/xci/src/zig/build/zig2.c: In function ‘unicode_utf8ToUtf16Le__103960’:
/home/xci/src/zig/build/zig2.c:1185794: warning: assignment to ‘const uint16_t (*)[16]’ {aka ‘const short unsigned int (*)[16]’} from incompatible pointer type ‘const uint16_t *’ {
aka ‘const short unsigned int *’} [-Wincompatible-pointer-types]
1185794 |    t26 = t25.ptr;
        |
[8/9] Building stage3
FAILED: stage3/bin/zig /home/xci/src/zig/build/stage3/bin/zig
cd /home/xci/src/zig && /home/xci/src/zig/build/zig2 build --prefix /home/xci/src/zig/build/stage3 --zig-lib-dir /home/xci/src/zig/lib -Dconfig_h=/home/xci/src/zig/build/config.h -
Denable-llvm -Doptimize=ReleaseFast -Dstrip -Dno-lib -Dno-langref -Dno-autodocs -Dtarget=native -Dcpu=native -Dversion-string=0.12.0-dev.1736+2252dcc50
/home/xci/src/zig/lib/std/debug.zig:2399:49: error: no field named 'code' in union 'c.netbsd.siginfo_t'
/home/xci/src/zig/lib/std/c/netbsd.zig:1148:30: note: union declared here
ninja: build stopped: subcommand failed.
```

siginfo_t indeed does not have a field 'code': 

```
typedef union siginfo {
        char    si_pad[128];    /* Total size; for future expansion */
        struct _ksiginfo _info;
} siginfo_t;
```

### Expected Behavior

Working zig compiler in build/stage3/bin/zig

This is with the latest commit to debug.zig; I have at present working

```
❯ zig version
0.12.0-dev.1664+8ca4a5240

```
built on the 20th of November.

**Comment by ci4ic4** [comment]
```
diff --git a/lib/std/debug.zig b/lib/std/debug.zig
index 236539807..4c0cfe5d6 100644
--- a/lib/std/debug.zig
+++ b/lib/std/debug.zig
@@ -2381,7 +2381,7 @@ fn handleSegfaultPosix(sig: i32, info: *const os.siginfo_t, ctx_ptr: ?*const any
     const addr = switch (native_os) {
         .linux => @intFromPtr(info.fields.sigfault.addr),
         .freebsd, .macos => @intFromPtr(info.addr),
-        .netbsd => @intFromPtr(info.info.reason.fault.addr),
+        //        .netbsd => @intFromPtr(info.info.reason.fault.addr),
         .openbsd => @intFromPtr(info.data.fault.addr),
         .solaris, .illumos => @intFromPtr(info.reason.fault.addr),
         else => unreachable,
```
lets me complete the build:
```
$ zig version
0.12.0-dev.1736+2252dcc50
$ uname -a
NetBSD narvi 10.99.10 NetBSD 10.99.10 (GENERIC64) #7: Sun Nov 12 05:54:51 GMT 2023  sysbuild@ymir.lorien.lan:/dumps/sy
sbuild/evbarm64/obj/home/sysbuild/src/sys/arch/evbarm/compile/GENERIC64 evbarm
```

**Comment by ianprime0509** [comment]
I don't have a NetBSD system to test, but it seems like the solution here would be to add a system conditional for `code` similar to the one already present for `addr`: https://github.com/ziglang/zig/blob/2252dcc508c53dc302ec1f919c4792a7f00ce125/lib/std/debug.zig#L2375-L2414 since, as you noted, the `siginfo_t` structure is different on NetBSD: https://github.com/ziglang/zig/blob/2252dcc508c53dc302ec1f919c4792a7f00ce125/lib/std/c/netbsd.zig#L1148-L1196

e.g.
```zig
const code = switch (native_os) {
    .netbsd => info.info.code,
    else => info.code,
};
```

**Comment by patterns** [comment]
Not sure whether I can include FreeBSD here or am required to report in a separate item. I have noticed the nightly source tarball has stopped building on my local FreeBSD 13.2 x86 machine (about 3 nights ago). It began failing before the 1717 file set; if I remember, the errors referenced a `compiler_rt.zig`. Tonight to eliminate my local env, I queued it on the Cirrus CI FreeBSD 14 image with the 1733 file set and the error references `Futex.zig`. Here is the log from the Cirrus CI tasks (https://api.cirrus-ci.com/v1/task/4999454562975744/logs/setup_zig.log). Also attaching the dot-cirrus.yml file that I use (it worked previously for the zig-0.11 nightly): https://github.com/patterns/pkcs1-verify/pull/17/commits/acb511670a12f6d87f6a1cd8599b2d866ce1b0e9
